### ISSUE TITLE
nil `virtual_path` unexpected by instrumentation when using view_context.render with absolute file path

### ISSUE THREAD

**Issue by PikachuEXE** [scoutapp/scout_apm_ruby#256]
This is triggered after upgrading to `2.4.23` from `2.4.22`
And the only change mentioned in change log is #255 

### Env
Ruby 2.5.4
Rails 5.2.2.1

### Stacktrace on Error Tracker from Production
![slacktrace_1](https://user-images.githubusercontent.com/1018543/54403973-e9405d80-470c-11e9-898d-a509e5bcbb34.png)
![slacktrace_2](https://user-images.githubusercontent.com/1018543/54403975-e9405d80-470c-11e9-90cd-ab36f9d71edb.png)

### `asset_to_string` helper
```ruby
def asset_to_string(name, html_safe: false)
  app = Rails.application
  result_string = if Rails.configuration.assets.compile
    app.assets.find_asset(name).to_s
  else
    absolute_file_path = File.join(
      Rails.public_path,
      "assets",
      app.assets_manifest.assets[name],
    )

    controller.view_context.render(
      file: absolute_file_path,
    )
  end

  return result_string unless html_safe

  # rubocop:disable Rails/OutputSafety
  result_string.html_safe
  # rubocop:enable Rails/OutputSafety
end
```

### Stacktrace in development
Reproduced via `controller.view_context.render(file: "#{Rails.public_path}/test.css")`
Test file is placed when testing
![dev_stacktrace_1](https://user-images.githubusercontent.com/1018543/54404377-37099580-470e-11e9-9107-a7f33e60af28.png)
![dev_stacktrace_2](https://user-images.githubusercontent.com/1018543/54404378-37099580-470e-11e9-85fe-d9365ebfa3c5.png)
![dev_stacktrace_3](https://user-images.githubusercontent.com/1018543/54404379-37099580-470e-11e9-9d8c-b2ab762e9087.png)
![dev_stacktrace_4](https://user-images.githubusercontent.com/1018543/54404380-37099580-470e-11e9-9eef-13e804f0c2a0.png)
![dev_stacktrace_5](https://user-images.githubusercontent.com/1018543/54404382-37a22c00-470e-11e9-8c2c-fbc11569de1e.png)
![dev_stacktrace_6](https://user-images.githubusercontent.com/1018543/54404383-37a22c00-470e-11e9-9b13-f5c2c4d79f50.png)
![dev_stacktrace_7](https://user-images.githubusercontent.com/1018543/54404384-37a22c00-470e-11e9-9e25-a8c877b42802.png)

**Comment by cschneid** [comment]
Thank you for the excellent report - It helped me see what's going on very quickly.  When I added the `prepend` version of the View instruments, they were supposed to be identical in behavior.

But, comparing the two code snippets:

```
# A
'#{args[0].virtual_path rescue "Unknown"}/Rendering'

# B
template_name = @template.virtual_path rescue "Unknown Partial"
layer_name = template_name + "/Rendering"
```

It appears that `.virtual_path` is legitimately returning nil (and not exceptioning) in the case of a raw render, so we end up with `nil + string` causing the error.

I'll have a patch for this out in a bit.

**Comment by cschneid** [comment]
@PikachuEXE released in 2.4.24. Thanks again!

**Comment by PikachuEXE** [comment]
Tested in dev and prod and it's working now
Thanks
